packages/core/src/components/button/button.css

```diff
@@ -176,65 +176,25 @@
 input[type="submit"].hds-button,
 .hds-button__label {
   font-weight: inherit;
-  line-height: 1.25em;
-  padding: var(--spacing-s);
+}
+
+.hds-button__label {
+  padding: 0 var(--spacing-2-xs);
 }
 
 input[type="submit"].hds-button {
   cursor: pointer;
-  padding: var(--spacing-s) var(--spacing-l);
 }
 
 /* submit input */
 input[type="submit"].hds-button:focus-visible {
   box-shadow: 0 0 0 var(--outline-gutter) var(--submit-input-focus-gutter-color), 0 0 0 calc(var(--outline-gutter) + var(--outline-width)) var(--focus-outline-color);
 }
 
-/* no icons */
-.hds-button__label:only-child {
-  margin: 0 var(--spacing-2-xs);
-}
-
 input[type="submit"].hds-button--small,
-.hds-button--small .hds-button__label {
-  line-height: var(--lineheight-s);
-  padding: var(--spacing-2-xs) var(--spacing-xs);
-}
-
-
-/* supplementary with right icon */
-.hds-button--supplementary .hds-button__label:first-child {
-  padding-right: var(--spacing-2-xs);
-}
-
-/* no icons */
-.hds-button--small .hds-button__label:only-child {
-  margin: 0 var(--spacing-xs);
-}
-
-/* supplementary with left icon */
-.hds-button--supplementary .hds-icon + .hds-button__label:last-child {
-  padding-left: var(--spacing-2-xs);
-}
-
-/* supplementary with both icons */
-.hds-button--supplementary .hds-icon + .hds-button__label:not(:last-child) {
-  padding-left: var(--spacing-2-xs);
-  padding-right: var(--spacing-2-xs);
-}
-
-/* SMALL */
 .hds-button--small {
-  padding: 0;
-}
-
-input[type="submit"].hds-button--small {
-  padding: var(--spacing-2-xs) var(--spacing-m);
-}
-
-/* both icons */
-.hds-button--small .hds-button__label:not(:first-of-type):not(:last-of-type) {
-  padding: var(--spacing-2-xs) var(--spacing-2-xs);
```

Comment:
I a bit concerned there has never been stories for all of these odd cases and therefore loki tests won't cover them and their removal 🤔

Reply:
These were replace by the `gap` making the css a lot more simpler